dataclasses_json/core.py

```diff
@@ -244,7 +244,16 @@ def _support_extended_types(field_type, field_value):
         res = (field_value
                if isinstance(field_value, UUID)
                else UUID(field_value))
-    elif _issubclass_safe(field_type, (int, float, str, bool)):
+    elif _issubclass_safe(field_type, bool):
+        # issubclass(bool, int) -> True
+        # thus values >1 will always convert to True in the next clause, unless intercepted
```

Comment:
I do not see >1 ints being set to true, or am I blind?

Reply:
Run code from the issue author with 0.5.14 and you will set the 1234 int field being set to True when running `from_dict`

Reply:
How is that caught by your if-statements below?

Reply:
It is not caught, we just do not allow the invocation to happen in the first place, if target type is bool